src/main/osd/osd_elements.c

```diff
@@ -601,6 +604,54 @@ static void osdElementArtificialHorizon(osdElementParms_t *element)
 
     element->drawElement = false;  // element already drawn
 }
+
+static void osdElementUpDownReference(osdElementParms_t *element){
+// Up/Down reference feature displays reference points on the OSD at Zenith and Nadir
+// The positions are absolute and immutable by the user for consistent operation
+    if(abs(attitude.values.pitch)>450){ //pitch is greater than +45 or less than -45
+        float earthUpinBodyFrame[3] = {-rMat[2][0], -rMat[2][1], -rMat[2][2]}; //transforum the up vector to the body frame
+
+        float thetaB; // pitch from body frame to up vector
+        int psiBSign; // psi from body frame to up vector
+        char buf[2]; // character buffer
+
+        if(attitude.values.pitch>0.0){ //nose down
+            thetaB = -asinf(earthUpinBodyFrame[2]) * (180.0f / M_PIf); // get theta
+            psiBSign = -1; // correct the sign
+            buf[0] = '-';
+            buf[1] = 0;
+        }
+        else{// nose up
+            thetaB = asinf(earthUpinBodyFrame[2]) * (180.0f / M_PIf); // get theta 
+            psiBSign = 1; // correct the sign
+            buf[0] = '+';
+            buf[1] = 0;
+        }
+
+        int psiB = psiBSign*asinf(earthUpinBodyFrame[1]) * (180.0f / M_PIf); // calculate the yaw from body to up/down vector
+
+//      Symbology Position Calculation 
+//
+//      ////// Y ////////
+//      OSD Pos  Row   ThetaB 
+//      Top    =  0    -55
+//      Mid    =  7      0
+//      Bottom = 15     55
+
+//      ////// X ////////
+//      OSD Pos Col   PsiB  
+//      Left   =  0    -59
+//      Mid    = 14      0
+//      Right  = 28     59
+
+        int posX = round(0.237288136*psiB+14.0); //col
```

Comment:
`scaleRangef(psiB, -M_PIf / 4, M_PIf / 4, 0, 28)` (using element size instead of 0, 28 and with correct angles) is easier to understand, but a bit slower. Not sure about rounding, using correct bounds may be enough.

Also with custom size, you should check for overflow (IMO one angle shall be limited by attitude.values.pitch, but second is pretty unlimited? -90..90 deg)